src/bp/core/server.ts

```diff
@@ -274,34 +278,48 @@ export default class HTTPServer {
   }
 
   async decodeExternalToken(externalToken): Promise<any | undefined> {
-    const { enabled, publicKey, audience, algorithm } = await this._getExternalAuthConfig()
+    const externalAuth = await this._getExternalAuthConfig()
 
-    if (!enabled) {
+    if (!externalAuth || !externalAuth.enabled) {
       return undefined
     }
 
+    const { publicKey, audience, algorithm, issuer } = externalAuth
+
     const [scheme, token] = externalToken.split(' ')
     if (scheme.toLowerCase() !== 'bearer') {
       return new Error(`Unknown scheme "${scheme}"`)
     }
 
     return Promise.fromCallback(cb => {
-      jsonwebtoken.verify(token, publicKey, { audience, algorithm }, (err, user) => {
+      jsonwebtoken.verify(token, publicKey, { issuer, audience, algorithm }, (err, user) => {
         cb(err, !err ? user : undefined)
       })
     })
   }
 
   @Memoize
-  private async _getExternalAuthConfig(): Promise<ExternalAuthConfig> {
+  private async _getExternalAuthConfig(): Promise<ExternalAuthConfig | undefined> {
     const botpressConfig = await this.configProvider.getBotpressConfig()
     const config = botpressConfig.pro.externalAuth
 
-    if (!config.publicKey && config.enabled) {
-      try {
-        config.publicKey = await this.ghostService.global().readFileAsString('/', 'key.pub')
-      } catch (error) {
-        this.logger.attachError(error).error(`Couldn't open public key file /data/global/key.pub`)
+    if (!config) {
+      return
+    }
+
+    if (config.enabled) {
```

Comment:
Joining (!config || !config.enabled) would remove the need for another nested if